main/src/library/Environment.flix

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Ramiro Calle
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Environment {
+  def _envMapToFlix(javaMap: ##java.util.Map): Map[String, String] & Impure =
+    import java.util.Map.entrySet();
+    import java.util.Set.iterator();
+    import java.util.Iterator.hasNext();
+    import java.util.Iterator.next();
+    import java.util.Map$Entry.getKey();
+    import java.util.Map$Entry.getValue();
+    let iter = javaMap.entrySet().iterator();
+    let iterDone = () -> not iter.hasNext();
+    let iterNext = () -> {
+      let entry = iter.next() as ##java.util.Map$Entry;
+      let key = entry.getKey() as String;
+      let value = entry.getValue() as String;
+      (key, value)
+    };
+    Iterator.toMap(Iterator(iterDone, iterNext))
```

Comment:
The point is more to avoid the use of `Iterator` (because we are making big changes to it, so its a bit unstable right now).